src/components/common/FullViewSnapshotButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { createSnapshot, getBillingProfiles, snapshotCreateDetails } from 'actions/index';
+import { TdrState } from 'reducers';
+import TerraTooltip from 'components/common/TerraTooltip';
+import { Button } from '@mui/material';
+import React, { Dispatch } from 'react';
+import {
+  BillingProfileModel,
+  DatasetModel,
+  SnapshotRequestContentsModelModeEnum,
+} from 'generated/tdr';
+import { Action } from 'redux';
+import { connect } from 'react-redux';
+import { now } from 'lodash';
+import { useOnMount } from '../../libs/utils';
+
+interface FullViewSnapshotButtonProps {
+  dispatch: Dispatch<Action>;
+  dataset: DatasetModel;
+  billingProfiles: Array<BillingProfileModel>;
+}
+
+function FullViewSnapshotButton({
+  dataset,
+  dispatch,
+  billingProfiles,
+}: FullViewSnapshotButtonProps) {
+  useOnMount(() => {
+    dispatch(getBillingProfiles());
+  });
+
+  const defaultBillingProfile = dataset.defaultProfileId;
+  // if user does not have permission on billing profile, disable button and show tooltip
+  const isDisabled = !billingProfiles.map((model) => model.id).includes(defaultBillingProfile);
+  const tooltipText = 'You do not have access to the billing profile associated with this dataset.';
+
+  const handleCreateFullViewSnapshot = () => {
+    const name = `Full_View_Snapshot_of_${dataset.name}_${now()}`;
```

Comment:
What is the use case for creating a snapshot with an automatically generated name? For many users, the snapshot name is used to keep track of a specific data release, and I believe that providing a custom name is an important part of that process. Have we validated the use case where we don't allow the users to name the snapshot?

Reply:
I believe that users will be able to enter their own name with this next ticket: https://broadworkbench.atlassian.net/browse/DT-1001